includes/types/CMB2_Type_Associated_Objects.php

```diff
@@ -91,22 +104,24 @@ public function render( $args = array() ) {
 			add_action( 'admin_footer', 'find_posts_div' );
 		}
 
+		$this->query_object_type = $this->field->options( 'query_object_type' );
+		$this->query_args = $this->field->options( 'query_args' );
 		$this->field->add_js_dependencies( 'cmb2-associated-objects' );
 
-		$this->query_args = (array) $this->field->options( 'query_args' );
-		$this->setup_query_args();
-
 		$filter_boxes = '';
-		// Check 'filter' setting
-		if ( $this->field->options( 'filter_boxes' ) ) {
-			$filter_boxes = '<div class="search-wrap"><input type="text" placeholder="' . sprintf( __( 'Filter %s', 'cmb' ), $this->post_type_labels ) . '" class="regular-text search" name="%s" /></div>';
-		}
+		// Set .has_thumbnail
+		$has_thumbnail = $this->field->options( 'show_thumbnails' ) ? ' has-thumbnails' : '';
+		$hide_selected = $this->field->options( 'hide_selected' ) ? ' hide-selected' : '';
+
 
 		// Check to see if we have any meta values saved yet
 		$attached = $this->field->escaped_value();
 		$attached = empty( $attached ) ? array() : (array) $attached;
 
+		$query = $this->get_query( $this->query_object_type );
```

Comment:
This is called 3 times. Is that necessary? Or can we just do `$this->query = $this->get_query(`... and use that throughout?